planner/core/rule_inject_extra_projection.go

```diff
@@ -205,5 +206,9 @@ func injectProjBelowSort(p PhysicalPlan, orderByItems []*ByItems) PhysicalPlan {
 	if origChildProj, isChildProj := childPlan.(*PhysicalProjection); isChildProj {
 		refine4NeighbourProj(bottomProj, origChildProj)
 	}
+
+	if topProj.Schema().Len() == 0 {
```

Comment:
Add a comment for when will this be true

Reply:
If the schema length is 0, is this whole plan tree below projection needed?